met-web/src/components/engagement/form/EngagementForm.tsx

```diff
@@ -111,27 +114,35 @@ const EngagementForm = () => {
 
         return Object.values(errors).some((isError: unknown) => isError);
     };
-    const handleCreateEngagement = () => {
-        const errorExists = validateForm();
 
-        if (!errorExists) {
-            handleCreateEngagementRequest({
+    const handleSaveEngagement = async (isPreview: boolean) => {
+        const hasErrors = validateForm();
+
+        if (hasErrors) {
+            return Promise.reject('Validation Errors');
```

Comment:
better than placing everything under this under an if statement if(!errorExists), maybe we can just return nothing here instead of rejecting a promise. Worried that might have to handle this promise rejection if we don't need it

Reply:
updated